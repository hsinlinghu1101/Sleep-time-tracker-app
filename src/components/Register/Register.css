.signup-form{
    border-style: dotted;
    width:80%;
    border-color:#f2f7ff ;
}

input, select {
    margin: 15px 0;
}
.des{
    font-size: 12px;
    width: 80%;
}

.btn{
    font-size: 16px;
    padding: 3px 3%;
    margin: 5% 3%;
    display: inline-block;
    border: solid #ffe867;    
    color:#f2f7ff ;
}
.register{
    background-color: #10316b;
}

div.btn{
    padding: 1px 3%;
}
#age{
    color: black;
}