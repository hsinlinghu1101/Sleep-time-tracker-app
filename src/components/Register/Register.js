import React from 'react'
import { Link } from 'react-router-dom'
import AuthApiService from '../../Services/auth-api-service'


export default class Register extends React.Component{
    static defaultProps ={
        onRegisterationSuccess:() => {}
    }

    state= { 
        error: null, 
    }


    handleNewUser=(event)=>{
        event.preventDefault();
        const { user_name, user_age, password} = event.target
        
        this.setState({ error:null })
        AuthApiService.postUser({
            user_name: user_name.value,
            user_age: Number(user_age.value),
            password: password.value
        })
       
        .then(user =>{
            user_name.value =''
            user_age.value=''
            password.value=''
            this.props.onRegisterationSuccess()
        })
        .catch(res =>{
            this.setState({
                error:res.error
            })
        })
    }

    render(){
        const { error } = this.state
        return(
            <main role="main">
      
                <h3>Create an Account</h3>
            
            <form className='signup-form' onSubmit={this.handleNewUser}>
            {error && <p>{error}</p>}
                <div>   
                  <label htmlFor="name">User Name: </label>
                  <input placeholder='Name' type="text" name='user_name' id='name' required/><br></br>
                  <label htmlFor="age">Age: </label>
                  <select name='user_age' id='age'  required>
                    <option id='age'>find your age</option>
                    <option id='age' value='1'>14-17 year old</option>
                    <option id='age' value='2'>18-64 years</option>
                    <option id='age' value='3'>above 65 years</option>
                  </select>
                </div>
                <div>
                  <label htmlFor="password">Password: </label>
                  <input type="password" name='password' id='password' placeholder="password"  required/>
                  <p className='des'>(Password must have at least 9 characters and contain 1 upper case, lower case, number and special character)</p>
                </div>
                
                <button className='btn register' type='submit'>Sign Up</button>
                <div className='btn'><Link to='/'>Cancel</Link></div>
            </form>
            <div className='end'></div>
            </main>
            )
    }
       
    
}