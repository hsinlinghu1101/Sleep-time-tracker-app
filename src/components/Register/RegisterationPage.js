import React from 'react'
import Register from './Register'
import './Register.css' 
export default class RegistrationPage extends React.Component {
    static defaultProps={
        history:{
            push:()=>{}
        }
    }

    handleRegistration=(user)=>{
        const {history} = this.props
        history.push('/login')
    }
    render(){
        return(
            <div>
                <Register onRegisterationSuccess={this.handleRegistration}/>
            </div>
        )
    }

}