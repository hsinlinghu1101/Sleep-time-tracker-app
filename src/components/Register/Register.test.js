import React from 'react';
import ReactDOM from 'react-dom';
import {MemoryRouter as Router} from 'react-router-dom'
import Register from './Register';
import renderer from 'react-test-renderer';

it('renders without crashing', ()=>{
    const div = document.createElement('div')

    ReactDOM.render(
        <Router><Register/></Router>
        , div)
        ReactDOM.unmountComponentAtNode(div)
    
})

it('renders the UI as expected', () => {
    const tree = renderer.create(<Router><Register/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
})