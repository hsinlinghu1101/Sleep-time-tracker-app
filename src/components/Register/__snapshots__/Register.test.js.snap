// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  role="main"
>
  <h3>
    Create an Account
  </h3>
  <form
    className="signup-form"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="name"
      >
        User Name: 
      </label>
      <input
        id="name"
        name="user_name"
        placeholder="Name"
        required={true}
        type="text"
      />
      <br />
      <label
        htmlFor="age"
      >
        Age: 
      </label>
      <select
        id="age"
        name="user_age"
        required={true}
      >
        <option
          id="age"
        >
          find your age
        </option>
        <option
          id="age"
          value="1"
        >
          14-17 year old
        </option>
        <option
          id="age"
          value="2"
        >
          18-64 years
        </option>
        <option
          id="age"
          value="3"
        >
          above 65 years
        </option>
      </select>
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password: 
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        required={true}
        type="password"
      />
      <p
        className="des"
      >
        (Password must have at least 9 characters and contain 1 upper case, lower case, number and special character)
      </p>
    </div>
    <button
      className="btn register"
      type="submit"
    >
      Sign Up
    </button>
    <div
      className="btn"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Cancel
      </a>
    </div>
  </form>
  <div
    className="end"
  />
</main>
`;
